ps_mbo.php

```diff
@@ -826,4 +971,129 @@ private function translateTab($tab, $languages)
+            $distributionApi = $this->get('mbo.cdc.client.distribution_api');
+
+            if (
+                $authenticationProvider instanceof AuthenticationProvider
+                 && $distributionApi instanceof Client
+            ) {
+                $distributionApi->setBearer($authenticationProvider->getMboJWT());
+                $distributionApi->unregisterShop();
+            }
+        } catch (Exception $exception) {
+            // Do nothing here, the exception is caught to avoid displaying an error to the client
+            // Furthermore, the operation can't be tried again later as the module is now disabled or uninstalled
+        }
+    }
+
+    /**
+     * Install configuration for each shop
+     *
+     * @return bool
+     */
+    private function installConfiguration()
+    {
+        $result = true;
+
+        $this->configurationList['PS_MBO_SHOP_ADMIN_UUID'] = Uuid::uuid4()->toString();
+
+        foreach (Shop::getShops(false, null, true) as $shopId) {
+            foreach ($this->configurationList as $name => $value) {
+                if (false === Configuration::hasKey($name, null, null, (int) $shopId)) {
+                    $result = $result && (bool) Configuration::updateValue(
+                            $name,
+                            $value,
+                            false,
+                            null,
+                            (int) $shopId
+                        );
+                }
+            }
+        }
+
+        return $result;
+    }
+
+    private function callServiceWithLockFile(string $method, array $params = [])
+    {
+        $lockFile = $this->moduleCacheDir . $method . '.lock';
+        try {
+            // If the module is installed via command line or somehow the ADMIN_DIR is not defined,
+            // we ignore the shop registration, so it will be done at any action on the backoffice
+            if (php_sapi_name() === 'cli' || !defined('_PS_ADMIN_DIR_')) {
+                throw new Exception();
+            }
+            /** @var Client $distributionApi */
+            $distributionApi = $this->get('mbo.cdc.client.distribution_api');
+            if (!method_exists($distributionApi, $method)) {
+                return;
+            }
+
+            /**
+             * @var AuthenticationProvider $authenticationProvider
+             */
+            $authenticationProvider = $this->get('mbo.cdc.distribution_authentication_provider');
+            $distributionApi->setBearer($authenticationProvider->getMboJWT());
+            $distributionApi->{$method}($params);
+
+            if (file_exists($lockFile)) {
+                unlink($lockFile);
+            }
+        } catch (Exception $exception) {
+            // Create the lock file
+            if (!file_exists($lockFile)) {
+                if (!is_dir($this->moduleCacheDir)) {
+                    mkdir($this->moduleCacheDir);
```

Comment:
You may need to set the `$recursive` parameter to `true` here in case on of the intermediate directory in the path is missing.